edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/edgedeployment/pvc/KubernetesPvcMapperTest.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test.Edgedeployment.Pvc
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using global::Docker.DotNet.Models;
+    using k8s.Models;
+    using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.Docker;
+    using Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment.Pvc;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Moq;
+    using Xunit;
+
+    using AgentModels = global::Microsoft.Azure.Devices.Edge.Agent.Docker.Models;
+    using KubernetesConstants = Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Constants;
+
+    [Unit]
+    public class KubernetesPvcMapperTest
+    {
+        static readonly ConfigurationInfo DefaultConfigurationInfo = new ConfigurationInfo("1");
+        static readonly IDictionary<string, EnvVal> EnvVarsDict = new Dictionary<string, EnvVal>();
+        static readonly DockerConfig Config1 = new DockerConfig("test-image:1");
+        static readonly AgentModels.HostConfig VolumeMountHostConfig = new AgentModels.HostConfig
+        {
+            Mounts = new List<AgentModels.Mount>
+            {
+                new AgentModels.Mount()
+                {
+                    Type = "volume",
+                    ReadOnly = true,
+                    Source = "a-volume",
+                    Target = "/tmp/volumea"
+                },
+                new AgentModels.Mount()
+                {
+                    Type = "volume",
+                    ReadOnly = false,
+                    Source = "b-volume",
+                    Target = "/tmp/volumeb"
+                }
+            }
+        };
+        static readonly Dictionary<string, string> DefaultLabels = new Dictionary<string, string>
+        {
+            [KubernetesConstants.K8sEdgeDeviceLabel] = KubeUtils.SanitizeLabelValue("device1"),
+            [KubernetesConstants.K8sEdgeHubNameLabel] = KubeUtils.SanitizeLabelValue("hostname")
+        };
+
+        [Fact]
+        public void EmptyDirMappingForVolume()
+        {
+            var config = new CombinedDockerConfig("image", new AgentModels.CreateContainerParameters(), Option.None<AuthConfig>());
+            config.CreateOptions.HostConfig = VolumeMountHostConfig;
+            var docker = new DockerModule("module1", "v1", ModuleStatus.Running, Core.RestartPolicy.Always, Config1, ImagePullPolicy.OnCreate, DefaultConfigurationInfo, EnvVarsDict);
+            var module = new KubernetesModule(docker, config);
+            var mapper = new KubernetesPvcMapper(string.Empty, string.Empty, 0);
+
+            var pvcs = mapper.CreatePersistentVolumeClaims(module, DefaultLabels);
+
+            Assert.True(pvcs.HasValue);
+            pvcs.ForEach(
```

Comment:
I think when you are expecting to be `Some` in tests you can use this pattern. It will make code more comprehansable.
```
var result = Foo(bar).OrDefault();

Assert.Equal("something", result.V1);
...
```
as opposite to 
```
var result = Foo(bar).OrDefault();

Assert.True(result.HasValue);
result.ForEach(
  r => {
    Assert.Equal("something", result.V1);
    ...
  });
```

Reply:
I'm expecting `Some`, I should test for `Some`. `OrDefault()` doesn't throw or directly reveal that the calling function returned `None`.  However, you're right, it's more readable to use the `OrDefault()` - I changed it to this, which I think is a good compromise.
```csharp
            Assert.True(pvcs.HasValue);
            var pvcList = pvcs.OrDefault();
            Assert.True(pvcList.Any());
```